engines/pink/objects/actors/lead_actor.cpp

```diff
@@ -0,0 +1,523 @@
+	if (_state == kReady || _state == kMoving) {
+		Actor *clickedActor = getActorByPoint(point);
+		if (clickedActor && isInteractingWith(clickedActor)) {
+			_audioInfoMgr.start(clickedActor);
+		}
+
+		if (_state == kMoving)
+			cancelInteraction();
+	}
+}
+
+void LeadActor::onMouseMove(const Common::Point point) {
+	if (_state != kPDA)
+		updateCursor(point);
+	else
+		_page->getGame()->getPdaMgr().onMouseMove(point);
+}
+
+void LeadActor::onMouseOverWithItem(const Common::Point point, const Common::String &itemName, CursorMgr *cursorMgr) {
+	_cursorMgr->setCursor(kHoldingItemCursor, point, itemName + kClickable);
+}
+
+void LeadActor::onMouseOver(const Common::Point point, CursorMgr *mgr) {
+	if (getInventoryMgr()->isPinkOwnsAnyItems())
+		_cursorMgr->setCursor(kClickableFirstFrameCursor, point, Common::String());
+	else
+		Actor::onMouseOver(point, mgr);
+}
+
+void LeadActor::onClick() {
+	if (_isHaveItem) {
+		assert(_state != kMoving);
+		_isHaveItem = false;
+		_nextState = kUndefined;
+	} else {
+		if (_state == kMoving)
+			cancelInteraction();
+		startInventory(0);
+	}
+}
+
+void LeadActor::onInventoryClosed(bool isItemClicked) {
+	_isHaveItem = isItemClicked;
+	_state = _stateBeforeInventory;
+	_stateBeforeInventory = kUndefined;
+	_page->pause(false);
+	forceUpdateCursor();
+}
+
+void LeadActor::onWalkEnd(const Common::String &stopName) {
+	State oldNextState = _nextState;
+	_state = kReady;
+	_nextState = kUndefined;
+	if (_recipient && oldNextState == kPlayingSequence) {
+		if (_isHaveItem)
+			sendUseClickMessage(_recipient);
+		else
+			sendLeftClickMessage(_recipient);
+	} else {  // on ESC button
+		Action *action = findAction(stopName);
+		assert(action);
+		setAction(action);
+	}
+}
+
+void LeadActor::onPDAClose() {
+	_page->initPalette();
+	_page->getGame()->getDirector()->loadStage();
+
+	_state = _stateBeforePDA;
+	if (_state != kInventory)
+		_page->pause(0);
+}
+
+bool LeadActor::isInteractingWith(Actor *actor) {
+	if (!_isHaveItem)
+		return actor->isLeftClickHandlers();
+
+	return actor->isUseClickHandlers(getInventoryMgr()->getCurrentItem());
+}
+
+void LeadActor::setNextExecutors(const Common::String &nextModule, const Common::String &nextPage) {
+	if (_state == kReady || _state == kMoving || _state == kPlayingSequence || _state == kInventory || _state == kPDA) {
+		_state = kPlayingExitSequence;
+		_page->getGame()->setNextExecutors(nextModule, nextPage);
+	}
+}
+
+void LeadActor::forceUpdateCursor() {
+	PinkEngine *vm =_page->getGame();
+	vm->getDirector()->update(); // we have actions, that should be drawn to properly update cursor
+	const Common::Point point = vm->getEventManager()->getMousePos();
+	updateCursor(point);
+}
+
+void LeadActor::updateCursor(const Common::Point point) {
+	switch (_state) {
+	case kReady:
+	case kMoving: {
+		Actor *actor = getActorByPoint(point);
+		InventoryItem *item = getInventoryMgr()->getCurrentItem();
+		if (_isHaveItem) {
```

Comment:
While not against our CFCs I'd personally prefer a bit more consistency with regards to one-line-brace-usage in this whole construct :)